sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,74 @@ export class EventHubClient {
    */
   constructor(
     host: string,
-    entityPath: string,
+    eventHubPath: string,
     credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials,
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
-      );
-    }
-
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    let connectionString;
+    let tokenProvider = {} as TokenProvider;
+    hostOrConnectionString = String(hostOrConnectionString);
 
-    if (!tokenProviderOrCredentials) {
-      throw new Error("Please provide either a token provider or a valid credentials object.");
-    }
+    if (typeof eventHubPathOrOptions === "string") {
+      let host = hostOrConnectionString;
+      const eventHubPath = eventHubPathOrOptions;
+      let sharedAccessKeyName = "defaultKeyName";
+      let sharedAccessKey = "defaultKeyValue";
+      if (!host) {
+        throw new Error(
+          "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+        );
+      }
+      if (!eventHubPath) {
+        throw new Error("Please provide a value for the 'entityPath' parameter.");
```

Comment:
If we reach inside this `if` block, then it means that user provided empty string for `eventHubPath` parameter. We do not throw error for empty queue name in ServiceBus when creating queue client. 

I believe when the first send/receive operation is carried out, we get an error from the service. Can you check what error we get from the service, if we remove this check? If that is a sensible error from the service, then we can remove this check.

Reply:
If user provide empty string for `eventHubPath` then we throw this error:
`Either provide "path" or the "connectionString": "${connectionString}", must contain EntityPath="<path-to-the-entity>".` when we call `EventHubConnectionConfig.create(connectionString);`

Reply:
Ah! ok.